specs/eventing/overview.md

```diff
@@ -1,97 +1,133 @@
-backing implementation of the _Channel_.
-
-Event selection on a _Channel_ is 1:N – a single _Channel_ may fan out to
-multiple _Subscriptions_.
-
-See [Kind: Channel](spec.md#kind-channel).
-
----
-
-_Navigation_:
-
-- [Motivation and goals](motivation.md)
-- **Resource type overview**
-- [Interface contracts](interfaces.md)
-- [Object model specification](spec.md)
+The Knative Eventing API provides primitives for two common event-processing
+patterns:
+
+- Topology-based event routing ([`messaging.knative.dev`](#messaging))
+
+- Content-based event routing ([`eventing.knative.dev`](#eventing))
+
+Knative Eventing does not directly specify mechanisms for other event-processing
+models, including multi-stage workflows, correlated request-reply, and
+sequential (windowed) event processing; these models could be built using the
+primitives provided by Knative, or Knative could deliver events to an external
+system that implements these models.
+
+In addition to the primitives needed to express the above patterns, Knative
+Eventing defines two [_interface contracts_](#interface-contracts) to allow
+connecting multiple types of Kubernetes objects as event senders and recipients
+to the core primitives.
+
+![Overview of objects](images/eventing-overview.svg)
+<!-- Generated from the following UML using PlantUML:
+@startuml
+' Remove shadows
+skinparam shadowing false
+' Set ordering
+left to right direction
+
+usecase Addressable <<Knative Service>>
+usecase Addressable AS A2 <<Kubernetes Service>>
+
+package "eventing.knative.dev" {
+agent Broker
+agent "Trigger" as T1
+agent "Trigger" as T2
+}
+
+Broker <-- T1 : broker
+Broker <-- T2 : broker
+Addressable <-u- T1 : subscriber
+
+package "messaging.knative.dev" {
+agent Channel
+agent "Subscription" as S1
+agent "Subscription" as S2
+}
+
+Channel <-- S1 :channel
+Channel <-- S2 :channel
+Addressable <-u- S2 :subscriber
+A2 <-u- S2 : reply
+@enduml
+-->
+
+## Interface Contracts
+
+In addition to the concrete types described below in the `messaging.knative.dev`
+and `eventing.knative.dev` API groups, Knative Eventing supports referencing
+objects in other API groups as destinations for event delivery. This is done by
+defining partial schemas which the other resources must support. The following
+interface contracts define resource fragments and partial schemas (required fields
+on an arbitrary API object) which form a basis for Knative Eventing.
+
+### Addressable
+
+**Addressable** resources expose a resource address (HTTP URL) in their `status`
+object. The URL is used as a destination for delivery of events to the resource;
+the exposed URL must implement the [data plane contract](data-plane.md) for
+receiving events.
+
+[**Broker**](#broker) and [**Channel**](#channel) both implement **Addressable**
+to receive events from other components.
+
+### Destination
+
+**Destination** is an interface (resource fragment) which is used consistently
```

Comment:
I had a [discussion with @slinkydeveloper about this](https://github.com/knative/specs/pull/24#discussion_r633297006):

> I think Addressable is a contract, because you can access an object (via the K8s apiserver), and process it based on unpacking the partial schema.
>
> You can't do this for a Destination, because Destination is not rooted at the base of the object. Instead, you'll find Destinations in several different places in spec, but they all work the same. I wanted to distinguish between duck typing and "common fragments like ObjectReference", because common fragments don't have the same ability to work across variant Kinds.

Destination is a different type of thing than Addressable (which is a duck type). This is more akin to a K8s ObjectReference -- a common type used in many places, but which can't be operated on abstractly in the same way as a duck type.